### PR TITLE
Add more DTOs to API

### LINKED ISSUE #11244: API should use a new "SessionInfoDto" type to avoid duplicate types for ClientCapabilities and BaseItem
Currently the OpenAPI specification contains a few duplicate types because we're not using a data transfer object for SessionInfo.

The SessionInfo is "leaked" from the API in a few places:
- SessionController.getSessions
- SessionsMessage (in a property)
- AuthenticationResult (in a property)

These places need to be changed to use a (new) SessionInfoDto type, that uses the proper Dto types for its members. An alternative fix would be to use the dto types in the existing SessionInfo.

### PR DESCRIPTION
**Pull request by Shadowghost**
**Changes**
*  Add `SessionInfoDto`

**Issues**
Fixes #11244

### CODE REVIEW

File: MediaBrowser.Model/Dto/SessionInfoDto.cs [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    /// Gets or sets the additional users.
+    /// </summary>
+    /// <value>The additional users.</value>
+    public IReadOnlyList<SessionUserInfo>? AdditionalUsers { get; set; }
+
+    /// <summary>
+    /// Gets or sets the client capabilities.
+    /// </summary>
+    /// <value>The client capabilities.</value>
+    public ClientCapabilities? Capabilities { get; set; }
```
Comment (nielsvanvelzen) [inline MediaBrowser.Model/Dto/SessionInfoDto.cs:29]: Should use the DTO variant

**Review by nielsvanvelzen — CHANGES_REQUESTED** [review]

**Review by nielsvanvelzen — APPROVED** [review]
✅ It does look like `ClientCapabilities` (the non-dto type) is also used in `DeviceInfo` so we may need to add a `DeviceInfoDto` as-well. Probably better to do that in a separate pull request.

File: src/Jellyfin.Extensions/Json/Converters/JsonDelimitedArrayConverter.cs [hunk 2]
```diff
@@ -75,7 +59,22 @@ protected JsonDelimitedArrayConverter()
         /// <inheritdoc />
         public override void Write(Utf8JsonWriter writer, T[]? value, JsonSerializerOptions options)
         {
-            throw new NotImplementedException();
+            writer.WriteStartArray();
+            if (value is not null && value.Length > 0)
```
Comment (crobibero) [inline src/Jellyfin.Extensions/Json/Converters/JsonDelimitedArrayConverter.cs:63]: If `value` is null we shouldn't write an empty array

File: Jellyfin.Api/Controllers/SessionController.cs [hunk 3]
```diff
@@ -56,66 +47,25 @@ public SessionController(
...
+            User.GetUserId(),
+            deviceId,
+            activeWithinSeconds,
+            controllableUserToCheck);
+
+        if (result.Count == 0)
         {
-            var minActiveDate = DateTime.UtcNow.AddSeconds(0 - activeWithinSeconds.Value);
-            result = result.Where(i => i.LastActivityDate >= minActiveDate);
+            return NotFound();
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/SessionController.cs:68]: Since this endpoint is searching for sessions an `Ok` is correct here and not `NotFound`

Comment (Shadowghost) [reply Jellyfin.Api/Controllers/SessionController.cs:68]: Agreed, but this just replicates the current behaviour. Should I change it?

Comment (crobibero) [reply Jellyfin.Api/Controllers/SessionController.cs:68]: Yes, I would change it

**Review by crobibero — APPROVED** [review]

File: src/Jellyfin.Extensions/Json/Converters/JsonDelimitedArrayConverter.cs [hunk 4]
```diff
@@ -75,7 +65,29 @@ protected JsonDelimitedArrayConverter()
...
-            throw new NotImplementedException();
+            if (value is not null)
+            {
+                writer.WriteStartArray();
+                if (value.Length > 0)
+                {
+                    var toWrite = value.Length - 1;
+                    foreach (var it in value)
+                    {
+                        writer.WriteStringValue(it!.ToString());
```
Comment (Bond-009) [inline src/Jellyfin.Extensions/Json/Converters/JsonDelimitedArrayConverter.cs:76]: Shouldn't you properly handle null values here?

Comment (Shadowghost) [reply src/Jellyfin.Extensions/Json/Converters/JsonDelimitedArrayConverter.cs:76]: How do you think they should be handled? `"null"` or `null` or dropped?

Comment (crobibero) [reply src/Jellyfin.Extensions/Json/Converters/JsonDelimitedArrayConverter.cs:76]: I think `null` would typically be dropped

File: Jellyfin.Api/Controllers/SessionController.cs (hunk 3 above)
Comment (crobibero) [reply Jellyfin.Api/Controllers/SessionController.cs:68]: @Shadowghost I think this was missed?

File: src/Jellyfin.Extensions/Json/Converters/JsonDelimitedArrayConverter.cs [hunk 5]
```diff
@@ -75,7 +66,39 @@ protected JsonDelimitedArrayConverter()
...
+                writer.WriteStartArray();
+                if (value.Length > 0)
+                {
+                    var toWrite = value.Length - 1;
+                    foreach (var it in value)
+                    {
+                        var wrote = false;
+                        if (it is not null)
+                        {
+                            writer.WriteStringValue(it!.ToString());
```
Comment (crobibero) [inline src/Jellyfin.Extensions/Json/Converters/JsonDelimitedArrayConverter.cs:80]:
```suggestion
                            writer.WriteStringValue(it.ToString());
```

**Comment by Shadowghost** [comment]
@crobibero the whole logic for selecting sessions including this codepiece was moved into `public IReadOnlyList<SessionInfoDto> GetSessions`

**Comment by crobibero** [comment]
> @crobibero the whole logic for selecting sessions including this codepiece was moved into `public IReadOnlyList<SessionInfoDto> GetSessions`

I'm referring to the endpoint returning NotFound if no results are in the query instead of just returning an empty list

File: Jellyfin.Api/Controllers/SessionController.cs (hunk 3 above)
Comment (Shadowghost) [reply Jellyfin.Api/Controllers/SessionController.cs:68]: It was intentionally reverted because of comments on Matrix.

**Review by crobibero — APPROVED** [review]

File: src/Jellyfin.Extensions/Json/Converters/JsonDelimitedArrayConverter.cs [hunk 6]
```diff
@@ -35,38 +37,27 @@ protected JsonDelimitedArrayConverter()
...
-                {
-                    if (parsedValues[i] is not null)
-                    {
-                        typedValues.SetValue(parsedValues[i], typedValueIndex);
-                        typedValueIndex++;
-                    }
-                }
-
-                return typedValues;
+                return [.. typedValues];
```
Comment (Bond-009) [inline src/Jellyfin.Extensions/Json/Converters/JsonDelimitedArrayConverter.cs:60]:
```suggestion
                return typedValues.ToArray();
```

File: src/Jellyfin.Extensions/Json/Converters/JsonDelimitedArrayConverter.cs [hunk 7]
```diff
@@ -75,7 +66,39 @@ protected JsonDelimitedArrayConverter()
...
+                        {
+                            writer.WriteStringValue(it.ToString());
+                            wrote = true;
+                        }
+
+                        if (toWrite > 0)
+                        {
+                            if (wrote)
+                            {
+                                writer.WriteStringValue(Delimiter.ToString());
```
Comment (Bond-009) [inline src/Jellyfin.Extensions/Json/Converters/JsonDelimitedArrayConverter.cs:88]: Not sure about the perf impact of doing `ToString` everytime :thinking: 
For now this should be fine but I'll do some benchmarks when I've got time.

**Review by Bond-009 — APPROVED** [review]
Just 1 small comment

**Pull request merged**